bqskit/runtime/worker.py

```diff
@@ -338,6 +347,10 @@ def _handle_cancel(self, addr: RuntimeAddress) -> None:
             if not t.is_descendant_of(addr)
         ]
 
+        # if there was an error earlier, raise it now
+        if error is not None:
+            raise error
```

Comment:
Why do you delay the raise? Shouldn't the system crash immediately, or is this to test something?

Reply:
The idea is to finish cancelling tasks as much as possible before letting the system crash.  Although I suppose if its gonna crash, the system isn't going to be in a great state anyway.  Should I just let it crash immediately?

Reply:
Yeah, let it crash immediately.